### DISCUSSION TITLE
Please make instructions more understandable

### DISCUSSION THREAD

**Discussion by TheWhiteWord** [mudler/LocalAI#1031] [General]
Hi, first of all thanks for the work you are putting into this. Is appreciated.
But here is the thing. AI is not anymore something that only coders are into. . I am trying my best to trust me, but the time I spent getting my MA did not allow me to become a coder as well.
I have tried following your instructions but I can not manage to make this work. Now I could tell you what the errors are, and start a new "report", making this even more difficult for everyone involved, but I think the best solution is actually to start from the roots of the problem. And I think it starts here:

Make sure to save that in the **root of the LocalAI folder** 
_(I did enough work to know this, but I bet you there are plenty that will not know what a root is. even just saying the LocalAI folder would help a ton)_ 
Then **lets spin up the docker**
 _(I get this too, but for how cool the terminology is, is not helpful: Run the docker program or smth will be way more helpful, especially considering that some people may just get by with their English. Also is very unclear if the purpose is to have to work within the program for the following instruction, or just have the program running in the background. If the is so even: 'make sure that the Docker is running in the background would help avoid confusion)_ 
 Run this in a CMD or BASH

docker-compose up -d --pull always

Now we are going to let that set up, once it is done, let's check to make sure our huggingface / localai galleries are working (**wait until you see the ready screen** to do this)
 _(I never got the "ready" message on screen: I got: 
 Network localai_default Created
 Container localai-api-1 Started 
 I would advice to adjust the instructions so that one knows what you mean)_

curl http://localhost:8080/models/available

(I have no idea where this command should go. I assume you are referring to the cmd. a slight adjustment would go a long way: in the same cmd window type:  curl http://localhost:8080/models/available)

I want to point out that I have lots of respect for people who like you do their job with passion and just for the sake to help the community grow. Is admirable. And I also understand that is normal that one that is used to work in this field my find these comments ridiculous, as it is basic knowledge for you. 
But that is the point: I don't want to become a coder, I have my own passions. And right now its on you to make it accessible, but in the future, when I will use this technology to implement it in my own field , I promise to put the same effort to make it accessible to you).

peace

**Comment by lunamidori5** [comment]
Yes I am the one whom made that part of the site, I am working super hard on making it more clear each day! Thank you for your input as I am getting ready to do a big change and add photos to it too! Ill keep all of this in mind! Thank you again!

**Reply by ReliablyAwkward** [reply]
Yes I too am in a similar boat, went through all this hassle of getting Docker and whatnot, but to no avail... here is a link to my recent post of the issue: https://github.com/go-skynet/LocalAI/discussions/1115

Nothing but respect indeed, thank you for all that you do.

**Reply by lunamidori5** [reply]
@ReliablyAwkward did you try this? - https://localai.io/howtos/easy-setup-docker-gpu/

**Reply by lunamidori5** [reply]
https://localai.io/howtos/easy-setup-full/ is also a great way to get started too!

**Reply by ReliablyAwkward** [reply]
So much appreciation for your response :)
...Just gave it a gander and am already seduced by Medusa, via paralysis by analysis, because it has copy paste codes assuming I know where to put them and then assuming I'm already in the correct directory :(

In my journey to achieve the end game, I have obtained on my device this week Docker desktop, Git, Pytorch, Python, WSL, Ubuntu and a slew of sub-rogues such as something titled Chocolatey, CMake, etc...All seemingly sadly in vain since despite cloning repositories and learning about dependencies, I have read the words "Not found" maybe a billion times this week :(

Only reason I was able to make any progress is at all is because I was using different AI to walk me through it step by step and I instructed them to treat me like an infant born yesterday, educating me as it guides while telling me what to expect after every micro baby step, but the all elusive memory problem kicked in and continuity began breaking, yada yada. I was hoping to solve this by being able to train my own overtime, in harmony with the plug-in's for the note taking it software Obsidian, "Co-pilot" & "Smart connections"

**Reply by lunamidori5** [reply]
@ReliablyAwkward if there is a step that is unclear please let me know so I can update the site and fix it!

**Reply by ReliablyAwkward** [reply]
I don't know where to put the commands nor am I confident that I will be in the right place before the commands are entered, regarding directory location, this was the problem I had when trying to install it on my own, then I would use AI for help, and it was basically saying I saved too many things in too many different directories and the paths were all criss-crossed. 

I have all the tools I listed above ready to be used, but I want to be Proactive about the situation, so I don't run into any nonsense (not referring to you, but just the multitude of guidance I have been garnering from different ai's) later like "Oh well when you saved the XYZ, you didn't not put it in the root file of 123, so now let's dig deep and find Dockerfile and then move it to the root of Blah blah blah blah"

I asked Claude2 To treat me like someone born yesterday, and alien, where not a single piece of terminology is familiar, everything is foreign, from scratch, and educate me while you guide me on the terminology and then inform me as what I should expect to see per microbaby step, and it did the best job of all but then it hits its message limit :(

I already cloned the repository and all the buzzwords, went through every command to confirm its location and made sure I was in the right directory, But no matter what I did every time I went to go start local AI it just kept saying not found, and every time I copy pasted it over they all said how strange it was, "it must be a problem on the providers end, delete it all and do it again".

Which I already did twice and it was the same result, which led me here, please help as I am chronically advocating for the future of offline Ai to everyone I speak with and want to finally convert my confidence to competence, I will be local A.I.'s number one ambassador, lol :)

(If by any chance you are in Los Angeles I will gladly pay for an in person pow wow, please help)

**Reply by lunamidori5** [reply]
@ReliablyAwkward I can help on discord for free using screen share if you like - https://discord.gg/uJAeKSAGDy

**Reply by ReliablyAwkward** [reply]
O my goodness that would be exquisite, yes, of course... I just followed your link and managed to sign in to my old account, when would be a splendid time for you? I am currently attempting to utilize a plugin for Obsidian called SmartConnections, which is taking some time to obtain all the embeddings for my 2million word vault, so I don't want to risk interrupting it with screenshare, since I've experienced lag using that in the past. But a designated time later at your convenience? And anything I should have pre-loaded like this "Docker" thing, or? 

-Nothing but respect🙏

**Reply by lunamidori5** [reply]
@ReliablyAwkward just ping me on the discord, I am up when ever you want!

**Reply by lunamidori5** [reply]
@ReliablyAwkward also just pushed a PR to update the site to make it way more clear on what to do, there is also this that will fully set up and run localai on a like chatgpt like window - https://deploy-preview-66--localai.netlify.app/howtos/easy-setup-full/

**Comment by ReliablyAwkward** [comment]
> @ReliablyAwkward just ping me on the discord, I am up when ever you want!
I don't know what "ping" means apparently, since I just tried messaging you but it says "Your message could not be delivered. This is usually because you don't share a server with the recipient or the recipient is only accepting direct messages from friends. "

**Reply by lunamidori5** [reply]
Yes please ping me (by doing the ``@`` before my name on discord or copy and paste this into discord ``<@277660602560675841>`` @ReliablyAwkward My DMs are locked so all chating takes place on the server <3

**Comment by ReliablyAwkward** [comment]
> Yes please ping me (by doing the `@` before my name on discord or copy and paste this into discord `<@277660602560675841>` @ReliablyAwkward My DMs are locked so all chating takes place on the server <3

I'm at Discord and tried the @ symbol at the beginning of your name but it just brings me to a little window with your profile decorated with anime, Only option I see there to communicate is via the messaging box at the bottom but you said that is disabled. Anyway, I am deeply curious about LocalA.I.'s embedding experience, does it have a limit on batch requests?

And I looked at the updated links you sent but it still raises questions alike, exactly which terminal are those codes to be inputted and in which directory should I be in before doing so, etc.

Here is my Discord name if that helps you find me instead " ReliableAF "

**Reply by lunamidori5** [reply]
I pinged you right here - https://discord.com/channels/1096914990004457512/1096914990004457515/1159239687471370320

**Comment by gurbindersingh** [comment]
I am a software dev (though new to AI) and even I am having issue settings this up. I think often there is just some crucial information missing for people new to the project. Unfortunately I don't have the time to rework the documentation (or the know-how about the project) but I'd be willing to proof-/beta-read if that helps improve the documentation.

**Reply by ReliablyAwkward** [reply]
Yes I suspect it's something akin to telling a third world country kid who wants a new pair of shoes "just order them on Amazon"...There are so many fundamentals missing in the guidance that the concept lands D.O.A., or like telling that same kid how easy it is to bake cookies "just get in your car and go to the store to buy ingredients, and then whip them together before throwing them in the oven"...kids just lookin' up at you like "what the hell is an oven? a gas bill? an address to receive a gas bill?" etc ..... Call me a dummy because I 1000% need "Installing LocalAI for dummies"

**Comment by lunamidori5** [comment]
@gurbindersingh if the how to pages are unclear, please let me know how to make them better. - https://localai.io/howtos/

**Reply by gurbindersingh** [reply]
Ok here are a few things I think can be improved (note that this is only for the pages I've read myself while trying to get this to work)

- The last line in the [Docker+GPU setup](https://localai.io/howtos/easy-setup-docker-gpu/) can be easily missed (as it happened to me). Either make it an admonition or a heading above that that says *Next steps* 
- And for the [The Easy Model Import tutorial](https://localai.io/howtos/easy-model-import-downloaded/) 
  - Clarify if all three files (YAML config and templates) are required (I assume the templates don't work without the config file)
  - In the tutorial it only says what to add to the template files, but not how it's relevant or what it does. There should be a short explanation on the syntax.
  - This line `In the "lunademo.yaml" file (If you want to see advanced yaml configs - [Link](https://localai.io/advanced/))` should be an admonition, it's easy to miss when just scanning the page, you want to make sure that the readers know they can go look for the full config options somewhere. You should also rephrase this to `For a full list of options see [advance setup](https://localai.io/howtos/easy-model-import-downloaded/)`
- The [Advance Setup](https://localai.io/howtos/easy-model-import-downloaded/)
  - Here it'd be best to provide a minimal set of configs or at least clarify what the required configs are
  - Describe what do all the configs do or where people can look it up (e.g. if it's Llama.cpp specific and it can probably be found in their docs)

These are just a few things that I noticed. Because as in the [discussion I opened](https://github.com/go-skynet/LocalAI/discussions/1139) for some reason the program won't load the config, and I have no clue why. Is it because I didn't create any template files and those are required? Or am I missing some required config option? It's difficult to say for someone not actively working on the project with just the information that is provided in the doc.

**Reply by gurbindersingh** [reply]
Now, while I do agree that the documentation should be rewritten here and there so that even those who have no experience with program should be able to follow it, I don't necessarily think that it should be written so that even a five year old can follow it. That's just not possible without bloating it by 30% or so. Instead I think in a lot of cases a simply rephrasing can do a lot.

For example, instead of saying `Then lets spin up the Docker run this in a CMD or BASH.` you can simply say `Then run the following command to restart the Docker container.` I think that is a lot clearer.

And this could also benefit from a little rewrite:
<img width="1047" alt="image" src="https://github.com/go-skynet/LocalAI/assets/26441580/0d2de5b2-6e34-4b71-9ffb-55da7b9ca89a">
 
Above the `curl` command specify simply write a sentence:
Now you can send HTTP requests to the API, e.g. using the `curl` command:
`<curl command here>`
